contracts/DefaultStrategy.sol

```diff
@@ -0,0 +1,517 @@
+        finalVotes = new uint256[](groupsVoted);
+
+        for (uint256 i = 0; i < groupsVoted; i++) {
+            finalGroups[i] = sortedGroups[i].group;
+            finalVotes[i] = votesPerGroup[i];
+        }
+
+        return (finalGroups, finalVotes);
+    }
+
+    /**
+     * @notice Distributes withdrawals from default strategy by computing the number of votes that
+     * should be withdrawn from each group, then calling out to
+     * `Account.scheduleWithdrawals`.
+     * @param withdrawal The amount of votes to withdraw.
+     * CELO.
+     * @dev The withdrawal distribution strategy is to:
+     * 1. Withdraw as much as possible from any deprecated groups.
+     * 2. If more votes still need to be withdrawn, try and have each validator
+     * group end up receiving the same amount of votes from the system. If a
+     * group already has less votes than the average of the total remaining
+     * votes, it will not be withdrawn from, and instead we'll try to evenly
+     * distribute between the remaining groups.
+     */
+    function distributeWithdrawalsDefaultStrategy(uint256 withdrawal)
+        external
+        onlyManager
+        returns (address[] memory finalGroups, uint256[] memory finalVotes)
+    {
+        address[] memory deprecatedGroupsWithdrawn;
+        uint256[] memory deprecatedWithdrawalsPerGroup;
+        uint256 numberDeprecatedGroupsWithdrawn;
+
+        (
+            deprecatedGroupsWithdrawn,
+            deprecatedWithdrawalsPerGroup,
+            numberDeprecatedGroupsWithdrawn,
+            withdrawal
+        ) = getDeprecatedGroupsWithdrawalDistribution(withdrawal);
+
+        address[] memory groupsWithdrawn;
+        uint256[] memory withdrawalsPerGroup;
+
+        (groupsWithdrawn, withdrawalsPerGroup) = getActiveGroupWithdrawalDistribution(withdrawal);
+
+        finalGroups = new address[](groupsWithdrawn.length + numberDeprecatedGroupsWithdrawn);
+        finalVotes = new uint256[](groupsWithdrawn.length + numberDeprecatedGroupsWithdrawn);
+
+        for (uint256 i = 0; i < numberDeprecatedGroupsWithdrawn; i++) {
+            finalGroups[i] = deprecatedGroupsWithdrawn[i];
+            finalVotes[i] = deprecatedWithdrawalsPerGroup[i];
+        }
+
+        for (uint256 i = 0; i < groupsWithdrawn.length; i++) {
+            finalGroups[i + numberDeprecatedGroupsWithdrawn] = groupsWithdrawn[i];
+            finalVotes[i + numberDeprecatedGroupsWithdrawn] = withdrawalsPerGroup[i];
+        }
+
+        return (finalGroups, finalVotes);
+    }
+
+    /**
+     * @notice Marks a group as votable for default strategy.
+     * @param group The address of the group to add to the set of votable
+     * groups.
+     * @dev Fails if the maximum number of groups are already being voted for by
+     * the Account smart contract (as per the `maxNumGroupsVotedFor` in the
+     * Election contract).
+     */
+    function activateGroup(address group) external onlyManager {
```

Comment:
Should this be `onlyManager` or `onlyOwner`? I.e., when the MultiSig decides to activate a new group, is there a reason a function on Manager needs to be called first to forward this, or should the MultiSig just call this function directly? Ditto for any similar functions here and in other contracts (may have missed similar situations in SpecificGroupStrategy?).

Reply:
I originally wanted to keep backward compatibility but it is not really necessary with these functions - refactored